frontend/control-center/src/pages/Catalog/index.tsx

```diff
@@ -28,14 +34,19 @@ const Catalog = (props: ConnectedProps<typeof connector>) => {
     setPageTitle(catalogPageTitle);
   }, []);
 
-  useEffect(() => {
-    setOrderedCatalogList(Object.values(catalogList).sort(sortByInstall));
-  }, [catalogList]);
+  useLayoutEffect(() => {
+    const sortedByInstalled = [...catalogList]
+      .filter((component: ComponentInfo) => component.installed && component.price !== ConnectorPrice.requestAccess)
+      .sort(sortByName);
+    const sortedByUninstalled = [...catalogList]
+      .filter((component: ComponentInfo) => !component.installed && component.price !== ConnectorPrice.requestAccess)
+      .sort(sortByName);
+    const sortedByAccess = [...catalogList]
+      .filter((component: ComponentInfo) => component.price === ConnectorPrice.requestAccess)
+      .sort(sortByName);
 
-  const sortByInstall = (a: ComponentInfo) => {
-    if (a.installed) return 1;
-    return -1;
-  };
+    setOrderedCatalogList(sortedByInstalled.concat(sortedByUninstalled).concat(sortedByAccess));
+  }, [catalogList]);
 
```

Comment:
I understand that we are manipulating the DOM on each change, but I don't see a difference between useEffect and useLayoutEffect in this case (and useEffect is more performant).